codegen/src/api/mod.rs

```diff
@@ -94,15 +94,25 @@ pub fn generate_runtime_api_from_url(
     should_gen_docs: bool,
     runtime_types_only: bool,
 ) -> Result<TokenStream2, CodegenError> {
-    // Fetch latest unstable version, if that fails fall back to the latest stable.
-    let bytes = match fetch_metadata_bytes_blocking(url, MetadataVersion::Unstable) {
-        Ok(bytes) => bytes,
-        Err(_) => fetch_metadata_bytes_blocking(url, MetadataVersion::Latest)?,
+    fn fetch_metadata(url: &Uri, version: MetadataVersion) -> Result<Metadata, CodegenError> {
+        let bytes = fetch_metadata_bytes_blocking(url, version)?;
+        Ok(Metadata::decode(&mut &bytes[..])?)
+    }
+
+    let metadata = match fetch_metadata(url, MetadataVersion::Unstable) {
```

Comment:
We shouldnt try to fetch unstable metadata now unless subxt has the unstable-metadata feature activated

one way to do this might be to expose from subxt eg a const boolean called UNSTABLE_METADATA that is true if the feature is enabled and false if it's not (via feature flag) and is doc(hidden).

this crate could then check that to decide whether to try and fetch unstable first.

Reply:
Actually that's no good because the crate doesn't depend on subxt. Let's just have it look for V15 and then V14, and optionally we can also add an "unstable_metadata" attr to enable it fetching unstable if we want :)